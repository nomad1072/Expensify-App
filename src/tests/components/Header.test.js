import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from '../../components/Header';

test('should render Headeer correctly', () => {
  const wrapper = new shallow(<Header />);
  expect(toJSON(wrapper)).toMatchSnapshot();
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('h1').text()).toBe('Expensify');

  // renderer.render(<Header />);
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
  // console.log(renderer.getRenderOutput());
});


