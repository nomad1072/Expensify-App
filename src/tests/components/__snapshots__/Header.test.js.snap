// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Headeer correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <header>
    <h1>
      Expensify
    </h1>
    <NavLink
      activeClassName="is-active"
      exact={true}
      to="/"
    >
      Dashboard
    </NavLink>
    <NavLink
      activeClassName="is-active"
      to="/create"
    >
      Create Expense
    </NavLink>
    <NavLink
      activeClassName="is-active"
      to="/help"
    >
      Help
    </NavLink>
  </header>,
  "nodes": Array [
    <header>
      <h1>
        Expensify
      </h1>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/"
      >
        Dashboard
      </NavLink>
      <NavLink
        activeClassName="is-active"
        to="/create"
      >
        Create Expense
      </NavLink>
      <NavLink
        activeClassName="is-active"
        to="/help"
      >
        Help
      </NavLink>
    </header>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Header />,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <header>
      <h1>
        Expensify
      </h1>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/"
      >
        Dashboard
      </NavLink>
      <NavLink
        activeClassName="is-active"
        to="/create"
      >
        Create Expense
      </NavLink>
      <NavLink
        activeClassName="is-active"
        to="/help"
      >
        Help
      </NavLink>
    </header>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header />,
}
`;
